main.js

```diff
@@ -126,44 +126,42 @@ class Covid19 extends utils.Adapter {
 
 				// Write Top 5
 				this.log.debug('Top 5 Countries : ' + JSON.stringify(values.slice(0, 5)));
-				const top_Arrary = values.slice(0, 5);
-				let count = 1;
-				for (const i in top_Arrary) {
-					if (count <= 5 ) {
-						let country = top_Arrary[i].country;
-
-						await this.extendObjectAsync('country_Top_5.' + count, {
-							type: 'channel',
-							common: {
-								name: 'Rank ' + count + ' : ' + country,
-							},
-							native: {},
-						});
-
-						country = country.replace(/\s/g, '_');
-						country = country.replace(/\./g, '');
-						this.log.debug('Country loop rank : ' + count + ' ' + JSON.stringify(country));
-						for (const y in values[i]) {
-							if (y !== 'countryInfo') {
-								await this.localCreateState('country_Top_5.' + count + '.' + y, y, values[i][y]);
-							} else {
-								// Only take the flag from country info
-								await this.localCreateState('country_Top_5.' + count + '.flag', 'flag', values[i][y].flag);
-							}
+				for (let position = 1; position <= 5; position++) {
+					const dataset = values[position - 1]; // start at 0
+					let country = dataset.country;
+
+					const channelName = 'country_Top_5.' + position;
+
+					await this.extendObjectAsync(channelName, {
+						type: 'channel',
+						common: {
+							name: 'Rank ' + position + ' : ' + country,
+						},
+						native: {},
+					});
+
+					country = country.replace(/\s/g, '_');
+					country = country.replace(/\./g, '');
+					this.log.debug('Country loop rank : ' + position + ' ' + JSON.stringify(country));
+					for (const property of Object.keys(dataset)) {
+						if (property !== 'countryInfo') {
+							await this.localCreateState(`${channelName}.${property}`, property, dataset[property]);
+						} else {
+							// Only take the flag from country info
+							await this.localCreateState(`${channelName}.flag`, 'flag', dataset[property].flag);
 						}
 					}
-					count = count + 1;
 				}
 
 
 				// Write continent information
-				if (continentsStats && this.config.getContinents === true) {
+				if (this.config.getContinents === true) {
```

Comment:
You initalize continentsStats at the start of the loop, so no need for that check.